classes/Telemetry.php

```diff
@@ -0,0 +1,353 @@
+					} else {
+						$conditions[ $target['target'] ] = 1;
+					}
+				}
+			}
+
+			// Add locations setting.
+			if ( isset( $location[ $settings['location'] ] ) ) {
+				$location[ $settings['location'] ] += 1;
+			} else {
+				$location[ $settings['location'] ] = 1;
+			}
+
+			// Add size setting.
+			if ( isset( $sizes[ $settings['size'] ] ) ) {
+				$sizes[ $settings['size'] ] += 1;
+			} else {
+				$sizes[ $settings['size'] ] = 1;
+			}
+
+			// Add opening sound setting.
+			if ( isset( $sounds[ $settings['open_sound'] ] ) ) {
+				$sounds[ $settings['open_sound'] ] += 1;
+			} else {
+				$sounds[ $settings['open_sound'] ] = 1;
+			}
+		}
+
+		$args = array(
+			// UID
+			'uid'              => self::get_uuid(),
+
+			// Language Info
+			'language'         => get_bloginfo( 'language' ), // Language
+			'charset'          => get_bloginfo( 'charset' ), // Character Set
+
+			// Server Info
+			'php_version'      => phpversion(),
+			'mysql_version'    => $wpdb->db_version(),
+			'is_localhost'     => self::is_localhost(),
+
+			// WP Install Info
+			'url'              => get_site_url(),
+			'version'          => Popup_Maker::$VER, // Plugin Version
+			'wp_version'       => get_bloginfo( 'version' ), // WP Version
+			'theme'            => $theme,
+			'active_plugins'   => $active_plugins,
+			'inactive_plugins' => array_values( $plugins ),
+
+			// Popup Metrics
+			'popups'           => $popups,
+			'popup_themes'     => $popup_themes,
+			'open_count'       => get_option( 'pum_total_open_count', 0 ),
+
+			// Popup Maker Settings
+			'block_editor_enabled'   => pum_get_option( 'gutenberg_support_enabled' ),
+			'bypass_ad_blockers'     => pum_get_option( 'bypass_adblockers' ),
+			'disable_taxonimies'     => pum_get_option( 'disable_popup_category_tag' ),
+			'disable_asset_cache'    => pum_get_option( 'disable_asset_caching' ),
+			'disable_open_tracking'  => pum_get_option( 'disable_popup_open_tracking' ),
+			'default_email_provider' => pum_get_option( 'newsletter_default_provider', 'none' ),
+
+			// Aggregate Popup Settings
+			'triggers'   => $triggers,
+			'cookies'    => $cookies,
+			'conditions' => $conditions,
+			'locations'  => $location,
+			'sizes'      => $sizes,
+			'sounds'     => $sounds,
+		);
+
+		return $args;
+	}
+
+	/**
+	 * Sends check_in data
+	 *
+	 * @param array $data Telemetry data to send.
+	 * @since 1.11.0
+	 */
+	public static function send_data( $data = array() ) {
+		self::api_call( 'check_in', $data );
+	}
+
+	/**
+	 * Makes HTTP request to our API endpoint
+	 *
+	 * @param string $action The specific endpoint in our API.
+	 * @param array $data Any data to send in the body.
+	 * @return array|bool False if WP Error. Otherwise, array response from wp_remote_post.
+	 * @since 1.11.0
+	 */
+	public static function api_call( $action = '', $data = array() ) {
+		$response = wp_remote_post( 'https://api.wppopupmaker.com/wp-json/pmapi/v2/' . $action, array(
```

Comment:
This endpoint is the new one your making correct?

Reply:
Yeah, that new endpoint is already built and all new fields are already to go. Though, that code isn't on the live site yet but will match this API call here.